src/app/static/src/js/components/runWorkflow/runWorkflowRun.vue

```diff
@@ -1,73 +1,197 @@
+                workflowName: "",
+                runMetadata: null,
+                changeLogTypeValue: "",
+                changeLogMessage: "",
+                error: "",
+                defaultMessage: "",
+                workflows: [],
+                workflowNameError: "",
+                /**
+                 * switchFeatureComponentOn variable should be removed when we are ready to implement v2
+                 * This variable was added because I didnt want to remove the code
+                 * created. This blocked section is meant to be implemented in v2.
+                 */
+                switchFeatureComponentOn: false
+            }
+        },
+        computed: {
+            showInstances() {
+                return !!this.runMetadata && this.runMetadata.metadata.instances_supported;
+            },
+            showChangeMessage: function () {
+                return this.runMetadata && this.runMetadata.metadata.changelog_types;
+            }
+        },
+        methods: {
+            getRunMetadata: function () {
+                api.get('/report/run-metadata')
+                    .then(({data}) => {
+                        this.runMetadata = data.data
+                        this.error = "";
+                        this.defaultMessage = "";
+                    })
+                    .catch((error) => {
+                        this.error = error;
+                        this.defaultMessage = "An error occurred while retrieving data";
+                    });
+            },
+            getWorkflows: function () {
+                api.get(`/workflows`)
+                    .then(({data}) => {
+                        this.workflows = data.data
+                        this.error = "";
+                        this.defaultMessage = "";
+                    })
+                    .catch((error) => {
+                        this.error = error
+                        this.defaultMessage = "An error occurred while retrieving previously run workflows";
+                    })
+            },
+            handleValidation: function () {
+                let valid = !!this.workflowName
+                if (this.workflows.find(workflow =>
+                    workflow.name.toLowerCase() === this.workflowName.toLowerCase())) {
+                    valid = false;
+                    this.workflowNameError = "Workflow name already exists, please rename your workflow."
+                }
+                this.$emit("valid", valid);
+            }
+        },
+        mounted() {
+            this.getRunMetadata();
+            this.getWorkflows();
+        },
+        watch: {
+            runMetadata() {
+                if (this.runMetadata && this.runMetadata.metadata.changelog_types) {
+                    this.changeLogTypeValue = this.runMetadata.metadata.changelog_types[0];
+                }
+
+                if (this.runMetadata && this.runMetadata.metadata.instances_supported) {
```

Comment:
If you do change the metadata management approach as suggested above, the initial selected instances will need to go onto the metadata via emitting an `update` event, so that those values will be used even if the user doesn't explicitly change them.